src/main/java/com/datadog/ddwaf/WafBuilder.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed
+ * under the Apache-2.0 License.
+ *
+ * This product includes software developed at Datadog
+ * (https://www.datadoghq.com/). Copyright 2025 Datadog, Inc.
+ */
+
+package com.datadog.ddwaf;
+
+
+
+import com.datadog.ddwaf.exception.AbstractWafException;
+import com.datadog.ddwaf.exception.InternalWafException;
+import com.datadog.ddwaf.exception.InvalidRuleSetException;
+import com.datadog.ddwaf.exception.UnclassifiedWafException;
+
+import java.util.Map;
+
+public final class WafBuilder {
+    /**
+     *  The ptr field holds the pointer to PWAddContext and managed by PowerWAF
+     */
+    private long ptr;     // KEEP THIS FIELD!
+    private boolean online;
+
+    public WafBuilder() {
+        this(null);
+    }
+
+    public WafBuilder(WafConfig config) {
+        online = true;
+        config = (config == null ? WafConfig.DEFAULT_CONFIG : config);
+        this.ptr = initBuilder(config);
+        if (ptr == 0 && config != WafConfig.DEFAULT_CONFIG) {
+            // in case the provided config does not work, we try with a default config again
+            this.ptr = initBuilder(WafConfig.DEFAULT_CONFIG);
+        }
+    }
+
+    public boolean addOrUpdateRuleConfig(String uniqueId, Map<String, Object> definition, RuleSetInfo[] infoRef) throws InvalidRuleSetException {
```

Comment:
I would suggest mirroring here the original ddwaf API: `add_or_update_config`, just adapted to our convention, so something like `WafUpdateResult addOrUpdateConfig(String path, Map<String, Object> configuration)`.

Not all configs are rules, this layer of abstraction is independent of the type of configuration. Also while the natural way in C of getting diagnostics is passing a pointer to the input arguments, on the Java side it'd make sense for this to be part of the result.

So result would be something like `WafConfigResult` with fields `success` and `diagnostics`.